jans-cedarling/cedarling/src/jwt/decoding_strategy/error.rs

```diff
@@ -0,0 +1,46 @@
+use super::key_service;
+use jsonwebtoken as jwt;
+
+/// Error type for JWT decoding_strategy operations.
+///
+/// This enum defines errors that can occur during the parsing, validation,
+/// and processing of JWTs, including issues with the key service and unsupported
+/// algorithms.
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
```

Comment:
Better to name `DecodingError` or something like that. JWT already has error And Same name may confuse.

Reply:
i always use fully qualified naming when referring to errors (e.g. `key_service::Error`) so it looks redundant when using `key_serivice::KeyServiceError`. but if it's confusing then i changed it here [b431062](https://github.com/JanssenProject/jans/pull/9999/commits/b431062461dd38e3ba1abb4d2f650b89432ccfe1)